app/src/main/java/uk/gov/android/securestore/SharedPrefsStore.kt

```diff
@@ -23,31 +23,37 @@ class SharedPrefsStore(
 ) : SecureStore {
     private val sharedPrefs = context.getSharedPreferences(configuration.id, Context.MODE_PRIVATE)
 
-    override suspend fun upsert(key: String, value: String): String {
+    override suspend fun upsert(key: String, value: String, context: FragmentActivity): String {
+        authenticator.init(context)
         return suspendCoroutine { continuation ->
             try {
                 val result = cryptoManager.encryptText(value)
                     .also { writeToPrefs(key, it) }
                 continuation.resumeWith(Result.success(result))
+                authenticator.close(context)
             } catch (e: GeneralSecurityException) {
                 throw SecureStorageError(e)
             }
         }
     }
 
-    override fun delete(key: String) {
+    override fun delete(key: String, context: FragmentActivity) {
         writeToPrefs(key, null)
+        authenticator.init(context)
         try {
             cryptoManager.deleteKey()
+            authenticator.close(context)
```

Comment:
could you move this line to a finally block please?

Reply:
Done, thank you